vaadin-button-flow-parent/vaadin-button-flow/src/main/resources/META-INF/resources/frontend/buttonFunctions.js

```diff
@@ -0,0 +1,10 @@
+window.Vaadin.Flow.button = {
+  initDisableOnClick: (target) => {
+    const disableEvent = function (target) {
+      if (target.getAttribute('disableOnClick')) {
+        target.setAttribute('disabled', 'true');
+      }
+    };
```

Comment:
Move this function outside the `window.Vaadin.Flow.button`-object so it doesn't get re-created on each call:

```js
function disableOnClickListener({ currentTarget: button }) {
  button.disabled = !!button.disableOnClick;
}
```